cortx_fix

ajay-paratmandali opened the pull request:
Signed-off-by: Ajay Paratmandali <ajay.paratmandali@seagate.com>

## Problem Statement
<pre>
  <code>
  Story Ref (if any):
    Push all cortx-1.0 fix to main branch
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  Yes
  </code>
</pre>
## Problem Description
<pre>
  <code>
    Your Problem decription here...
       In past we have done many fix in cortx-1.0 but not commited to main branch.
Creating PR to push all privious fix to main branch.
  </code>
</pre>
## Solution
<pre>
  <code>
    Your Problem solution here...
Create saparate PR. All in cortx-1.0 should present in main branch.
But all main branch not need to be in cortx-1.0

- pcswrap change comment for improper word
- resource fix from cortx
- sync build with cortx
- logrotate sync from cortx
- improve cli from cortx
- Add pcs cluster managment class
- Support bundle
- metadata fix

  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
    Unit Testing details here...
    "No unit testing was done"
     Tested with RPM
  </code>
</pre>

ajaysrivas commented:
Check the Unit test cases section. If there is no unit tests done, say that "No unit testing was done".

indrajitzagade commented:
> 
> 
> Signed-off-by: Ajay Paratmandali [ajay.paratmandali@seagate.com](mailto:ajay.paratmandali@seagate.com)
> ## Problem Statement
> 
>   ```
> 
>   Story Ref (if any):
>     Push all cortx-1.0 fix to main branch
>   
> ```
> 
> 
> ## Unit testing on RPM done
> 
>   ```
> 
>   Yes
>   
> ```
> 
> 
> ## Problem Description
> 
>   ```
> 
>     Your Problem decription here...
>        In past we have done many fix in cortx-1.0 but not commited to main branch.
> Creating PR to push all privious fix to main branch.
>   
> ```
> 
> 
> ## Solution
> 
>   ```
> 
>     Your Problem solution here...
> Create saparate PR. All in cortx-1.0 should present in main branch.
> But all main branch not need to be in cortx-1.0
> 
> - pcswrap change comment for improper word
> - resource fix from cortx
> - sync build with cortx
> - logrotate sync from cortx
> - improve cli from cortx
> - Add pcs cluster managment class
> 
>   
> ```
> 
> 
> ## Unit Test Cases
> 
>   ```
> 
>     Unit Testing details here...
>   
> ```

@ajay-paratmandali  Can you please list the PR numbers as well for these changes in cortx-1.0 branch?

indrajitzagade commented on ha/resource/sspl:
```diff
@@ -152,7 +152,7 @@ stateful_demote() {
 
 stateful_promote() {
-    sleep 5
+    sleep 8
```
I think we can restore this sleep back to 5 seconds and even less than 5 seconds, since the salt-calls in sspl code have been converted to using provisioner APIs

cc @saurabhkhanvilkar @Madhura-08

ajay-paratmandali replied:
@indrajitzagade i think we can take this as separate pr as it is only to port to main.
i am creating issue for this.

ajay-paratmandali replied:
https://github.com/Seagate/cortx-ha/issues/206

saurabhkhanvilkar replied:
@indrajitzagade i remember @Madhura-08 had tried that and going below 5 was still not working even after her changes... i think let's keep it 8 for now.

ajay-paratmandali replied:
@saurabhkhanvilkar  i have created separate issue for this we can check it separately

This pr only to port from cortx to main

we can disscuss more on sleep time
Issue: https://github.com/Seagate/cortx-ha/issues/206

indrajitzagade commented:
> 
> 
> Signed-off-by: Ajay Paratmandali [ajay.paratmandali@seagate.com](mailto:ajay.paratmandali@seagate.com)
> ## Problem Statement
> 
>   ```
> 
>   Story Ref (if any):
>     Push all cortx-1.0 fix to main branch
>   
> ```
> 
> 
> ## Unit testing on RPM done
> 
>   ```
> 
>   Yes
>   
> ```
> 
> 
> ## Problem Description
> 
>   ```
> 
>     Your Problem decription here...
>        In past we have done many fix in cortx-1.0 but not commited to main branch.
> Creating PR to push all privious fix to main branch.
>   
> ```
> 
> 
> ## Solution
> 
>   ```
> 
>     Your Problem solution here...
> Create saparate PR. All in cortx-1.0 should present in main branch.
> But all main branch not need to be in cortx-1.0
> 
> - pcswrap change comment for improper word
> - resource fix from cortx
> - sync build with cortx
> - logrotate sync from cortx
> - improve cli from cortx
> - Add pcs cluster managment class
> 
>   
> ```
> 
> 
> ## Unit Test Cases
> 
>   ```
> 
>     Unit Testing details here...
>   
> ```

Also, @ajay-paratmandali  I do not see changes for https://github.com/Seagate/cortx-ha/pull/156 here?

ajay-paratmandali commented:
`Also, @ajay-paratmandali I do not see changes for #156 here?`

Yes this change already in main branch
There are some change i have added form cortx branch are in
 ha/core/node/replacement/refresh_context.py
file

consul watcher file already present in main and no change need for setup.

ajay-paratmandali commented:
`@ajay-paratmandali Can you please list the PR numbers as well for these changes in cortx-1.0 branch?`

- pcswrap change comment improvement and remove word like backlist
- resource fix from cortx
- sync build with cortx
- logrotate sync from cortx
- improve cli from cortx
- Add pcs cluster managment class
- Support bundle
- metadata fix

Few PR 
iem_fix: Fix restart operation in resource agent : # 204
metadata: Fix return code for metadata: # 200
Added new command support_bundle #197
Time improve #183
add_node_fix: Fix add node to cluster #182
Fix sleep time for sspl ra #177
consul_dependency_fix: Fix consul dependency #156
failback: Add support for failback #147

Added their fix but setup not changed so, fix will be there but pcs content will be disabled

indrajitzagade commented on ha/pcswrap/tests/status-xml-plain-resources.xml:
```diff
@@ -80,10 +80,6 @@
     <bans>
-<!--
-# TODO: Replace with non-offensive terms when
-# https://bugs.clusterlabs.org/show_bug.cgi?id=5437 is resolved
```
I hope we have a valid reason to remove this.

ajay-paratmandali replied:
i think it is needed i have not seed in cortx1.0 
but had discussion with @ArchanaLimaye 
Added this back
